### DISCUSSION TITLE
User data for hitboxes

### DISCUSSION THREAD

**Discussion by ufrshubham** [flame-engine/flame#1901] [Ideas]
I was wondering if it would be a good idea to add an `userData` property to hitboxes which then gets passed down to the collision callbacks. This would be similar to Fogre2d's user data.

Current implementation provides `other` as the immediate parent casted to PositionComponent. This works for most cases, but when using CompositeHitbox, other turns out to be `CompositeHitbox` instead of the actual parent.

**Comment by spydon** [comment]
I'm not a big fan of user data that is untyped, but the thing with the `CompositeHitbox` is a bug so good that you discovered that! You can of course already get the real parent by just calling `.parent` on the `CompositeHitbox` as a work around meanwhile.

**Reply by ufrshubham** [reply]
Yeah, I am using `.parent` for now. Was going to report it as an issue but wanted to confirm if we just want to fix it for CompositeHitbox, or do we want to go the `userData` way. But I agree that adding untyped is not worth it. I'll open an issue for the `CompositeHitbox` later.
